### PR TITLE
Remove Smart Sensors

### PR DESCRIPTION
**Pull request by ashb**
It was deprecated in 2.3 and since it is experimental we can now remove
it -- Deferrable operators is a much more efficient pattern to achieve
the same feature.

### CODE REVIEW

**Review by potiuk — COMMENTED** [review]
Another -1700 lines of code :)

**Review by eladkal — COMMENTED** [review]
There are some more refrences in docs:

`sensors.rst`:
https://github.com/apache/airflow/blob/df143aee8d9e7e0089b747bdd27addf63bb4962f/docs/apache-airflow/concepts/sensors.rst#L31

`deferring.rst`:
https://github.com/apache/airflow/blob/cabbf616bbc3139b1ce793c9daa20f13c82eb287/docs/apache-airflow/concepts/deferring.rst#L176

File: newsfragments/25507.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Removal of experimental Smart Sensors
+
+Smart Sensors  were added in 2.0 and deprecated in favour of Deferable operators in 2.2 and have now been removed.
```
Comment (jedcunningham) [inline newsfragments/25507.rst]:
```suggestion
Smart Sensors were added in 2.0 and deprecated in favour of Deferable operators in 2.2 and have now been removed.
```

(nit)

This should also be `25507.significant.rst`.

Comment (ashb) [reply newsfragments/25507.rst]: My other question about newsfragments if we should have any formatting in them (lists, headings etc?)

**Review by eladkal — APPROVED** [review]
LGTM

File: newsfragments/25507.rst (hunk 1 above)
Comment (jedcunningham) [reply newsfragments/25507.rst]: "should", no. "can", for significant yep!

Comment (jedcunningham) [reply newsfragments/25507.rst]: A little longer answer: Non-significant can only have a single line. Significant can be a single line, or multi-line rst.

**Review by jedcunningham — APPROVED** [review]

File: newsfragments/25507.rst (hunk 1 above)
Comment (jedcunningham) [reply newsfragments/25507.rst]: PR to improve the precommit: #25536

**Pull request merged**